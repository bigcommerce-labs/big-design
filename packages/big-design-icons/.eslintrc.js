module.exports = require('@bigcommerce/configs/eslint');
