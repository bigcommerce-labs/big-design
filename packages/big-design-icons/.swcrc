{
  "jsc": {
    "externalHelpers": true,
    "parser": {
      "syntax": "typescript",
      "tsx": true
    }
  },
  "module": {
    "type": "es6"
  },
  "env": {}
}
