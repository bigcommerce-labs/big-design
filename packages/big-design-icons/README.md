# BigDesign Icons

BigDesign Icons as React components.

### Documentation

You can find documentation, list of icons, and examples on our [docs page](https://bigcommerce.github.io/big-design/?path=/story/icons--overview).

### Quick start guide

Add BigDesign Icons and styled-components to your project using `npm`:

```
npm install @bigcommerce/big-design-icons styled-components
```

or with `yarn`:

```
yarn add @bigcommerce/big-design-icons styled-components
```

Import any icon component and use it anywhere in your app.

```jsx
import { StoreIcon } from '@bigcommerce/big-design-icons';

...

<App>
  <StoreIcon />
</App>
```
