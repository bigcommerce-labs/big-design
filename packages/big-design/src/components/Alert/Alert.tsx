import { CheckCircleIcon, CloseIcon, ErrorIcon, WarningIcon } from '@bigcommerce/big-design-icons';
import React, { memo, FunctionComponent, HTMLAttributes } from 'react';

import { excludePaddingProps, MarginProps } from '../../mixins';
import { Flex } from '../Flex';
import { H4 } from '../Typography';

import { StyledAlert, StyledButton } from './styled';

export interface AlertProps extends HTMLAttributes<HTMLElement>, MarginProps {
  iconLeft?: React.ReactChild;
  header?: string;
  variant?: 'danger' | 'primary' | 'success' | 'warning';
  onCloseButtonClick?(): void;
}

export const DEFAULT_VARIANT = 'primary';

const RawAlert: FunctionComponent<AlertProps> = memo(props => {
  const filteredProps = excludePaddingProps(props);
  const { children, iconLeft, header, onCloseButtonClick, variant, ...restProps } = filteredProps;

  const renderIcon = () => {
    if (iconLeft) {
      return iconLeft;
    }

    switch (variant) {
      case 'success':
        return <CheckCircleIcon color={variant} size="large" />;
      case 'warning':
        return <WarningIcon color={variant} size="large" />;
      case 'danger':
      case 'primary':
        return <ErrorIcon color={variant} size="large" />;
    }
  };

  return (
    <StyledAlert
      {...restProps}
      backgroundColor="white"
      borderRadius="normal"
      flexDirection={{ mobile: 'row' }}
      padding={{ mobile: 'xSmall', tablet: 'small' }}
      shadow="raised"
      variant={variant}
    >
      <Flex.Item flexShrink={0} paddingRight="small">
        {renderIcon()}
      </Flex.Item>
      <Flex.Item flexGrow={1}>
        {header && <H4 marginBottom="none">{header}</H4>}
        {children}
      </Flex.Item>
      {onCloseButtonClick && (
        <Flex.Item flexShrink={0} paddingLeft="small">
          <StyledButton variant="subtle" onClick={onCloseButtonClick} iconOnly={<CloseIcon size="medium" />} />
        </Flex.Item>
      )}
    </StyledAlert>
  );
});

export const Alert: FunctionComponent<AlertProps> = ({ className, style, ...props }) => <RawAlert {...props} />;

Alert.displayName = 'Alert';
Alert.defaultProps = {
  variant: DEFAULT_VARIANT,
};
