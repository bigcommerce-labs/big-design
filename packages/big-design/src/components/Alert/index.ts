import { AlertProps as _AlertProps } from './Alert';

export { Alert } from './Alert';
export type AlertProps = _AlertProps;
