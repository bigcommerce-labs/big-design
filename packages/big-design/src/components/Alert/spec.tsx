import { DeleteIcon } from '@bigcommerce/big-design-icons';
import { theme } from '@bigcommerce/big-design-theme';
import { fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';

import { Alert } from './Alert';

test('render Alert', () => {
  const { container } = render(
    <Alert>
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur officia ex. Ipsum eiusmod
      fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse deserunt nostrud ipsum id adipisicing enim velit
      labore. Nulla exercitation laborum laboris Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Alert>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('does not forward styles', () => {
  const { container } = render(
    <Alert style={{ background: 'red' }}>
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur officia ex. Ipsum eiusmod
      fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse deserunt nostrud ipsum id adipisicing enim velit
      labore. Nulla exercitation laborum laboris Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Alert>,
  );

  expect(container.getElementsByClassName('test').length).toBe(0);
  expect(container.firstChild).not.toHaveStyle('background: red');
});

test('renders a header and action', () => {
  const onClick = jest.fn();

  const { getByRole } = render(
    <Alert header="Test Header" onCloseButtonClick={onClick}>
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur officia ex. Ipsum eiusmod
      fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse deserunt nostrud ipsum id adipisicing enim velit
      labore. Nulla exercitation laborum laboris Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Alert>,
  );

  const header = getByRole('heading');

  expect(header).toBeInTheDocument();
  expect(header.textContent).toBe('Test Header');

  const actionButton = getByRole('button');

  expect(actionButton).toBeInTheDocument();
  expect(actionButton).toMatchSnapshot();
});

test('action options get forwarded to button', () => {
  const onClick = jest.fn();

  const { getByRole } = render(
    <Alert header="Test Header" onCloseButtonClick={onClick}>
      Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur officia ex. Ipsum eiusmod
      fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse deserunt nostrud ipsum id adipisicing enim velit
      labore. Nulla exercitation laborum laboris Lorem irure sit esse nulla mollit aliquip consectetur velit
    </Alert>,
  );

  fireEvent.click(getByRole('button'));

  expect(onClick).toHaveBeenCalled();
});

test('forwards data attributes', () => {
  const { getByTestId } = render(
    <Alert header="Test Header" data-testid="alert">
      Dolore proident eiusmod sint est enim laboris
    </Alert>,
  );

  const alert = getByTestId('alert');

  expect(alert).toBeInTheDocument();
});

test('ignores padding props', () => {
  const { getByTestId } = render(
    // @ts-ignore - ignoring since paddingRight is not a valid prop
    <Alert header="Test Header" data-testid="alert" paddingRight="xxxLarge">
      Dolore proident eiusmod sint est enim laboris
    </Alert>,
  );

  const alert = getByTestId('alert');

  expect(alert).not.toHaveStyle(`padding-right: ${theme.spacing.xxxLarge}`);
});

test('support margin props', () => {
  const { getByTestId } = render(
    // @ts-ignore - ignoring since paddingRight is not a valid prop
    <Alert header="Test Header" data-testid="alert" marginBottom="xxxLarge">
      Dolore proident eiusmod sint est enim laboris
    </Alert>,
  );

  const alert = getByTestId('alert');

  expect(alert).toHaveStyle(`margin-bottom: ${theme.spacing.xxxLarge}`);
});

test('render custom icon left alert', () => {
  const { container } = render(<Alert iconLeft={<DeleteIcon />}>Dolore proident eiusmod sint est enim laboris</Alert>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render success alert', () => {
  const { container } = render(<Alert variant="success">Dolore proident eiusmod sint est enim laboris</Alert>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render danger alert', () => {
  const { container } = render(<Alert variant="danger">Dolore proident eiusmod sint est enim laboris</Alert>);

  expect(container.firstChild).toMatchSnapshot();
});

test('render warning alert', () => {
  const { container } = render(<Alert variant="warning">Dolore proident eiusmod sint est enim laboris</Alert>);

  expect(container.firstChild).toMatchSnapshot();
});
