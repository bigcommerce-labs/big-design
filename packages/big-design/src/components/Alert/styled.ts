import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css, DefaultTheme, StyledComponent } from 'styled-components';

import { getBorderStyle } from '../../utils';
import { Grid } from '../Grid';
import { Link } from '../Link';
import { TextProps } from '../Typography';
import { StyleableH4, StyleableSmall } from '../Typography/private';

import { AlertProps } from './Alert';

export const StyledAlert = styled(Grid)<AlertProps>`
  ${({ theme }) => theme.shadow.floating}

  animation: ${({ theme }) => theme.keyframes.fadeIn} .5s ease-in-out;
  background-color: ${({ theme }) => theme.colors.white};
  grid-gap: ${({ theme }) => theme.spacing.small};
  max-width: ${({ theme }) => theme.helpers.remCalc(456)};
  padding: ${({ theme }) => theme.spacing.small};
  position: fixed;
  right: ${({ theme }) => theme.spacing.medium};
  top: ${({ theme }) => theme.spacing.medium};
  z-index: ${({ theme }) => theme.zIndex.fixed};

  ${({ onClose }) =>
    onClose
      ? css`
          grid-template-areas: 'icon messages close';
          grid-template-columns: ${({ theme }) =>
            `${theme.spacing.xLarge} 1fr ${theme.spacing.large}`};
        `
      : css`
          grid-template-areas: 'icon messages';
          grid-template-columns: ${({ theme }) => `${theme.spacing.xLarge} 1fr`};
        `}

  ${({ theme, type }) => type && getBorderStyle(type, theme)};
`;

export const StyledHeader = styled(StyleableH4)`
  line-height: ${({ theme }) => theme.spacing.medium};
  margin-bottom: ${({ theme }) => theme.spacing.xxSmall};
`;

export const StyledMessageItem = styled(StyleableSmall).attrs({ as: 'span' })`
  color: ${({ theme }) => theme.colors.secondary70};
  vertical-align: middle;
` as StyledComponent<'span', DefaultTheme, TextProps>;

export const StyledLink = styled(Link)`
  font-size: ${({ theme }) => theme.typography.fontSize.small};
  vertical-align: middle;
`;

StyledAlert.defaultProps = { theme: defaultTheme };
StyledHeader.defaultProps = { theme: defaultTheme };
StyledMessageItem.defaultProps = { theme: defaultTheme };
StyledLink.defaultProps = { theme: defaultTheme };
