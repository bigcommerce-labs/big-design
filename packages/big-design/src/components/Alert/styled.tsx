import { addValues, theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled from 'styled-components';

import { StyleableButton } from '../Button/private';
import { Flex, FlexProps } from '../Flex';

import { AlertProps, DEFAULT_VARIANT } from './Alert';

export const StyledAlert = styled(Flex)<FlexProps & Pick<AlertProps, 'variant'>>`
  border-left-width: ${({ theme }) => theme.spacing.xxSmall};
  border-left-style: solid;
  border-left-color: ${({ theme, variant }) => variant && theme.colors[variant]};
`;

export const StyledButton = styled(StyleableButton)`
  color: ${({ theme }) => theme.colors.secondary70};
  height: ${({ theme }) => addValues(theme.spacing.xLarge, theme.spacing.xxSmall)};
  width: auto;
`;

StyledAlert.defaultProps = {
  theme: defaultTheme,
  variant: DEFAULT_VARIANT,
};
StyledButton.defaultProps = { theme: defaultTheme };
