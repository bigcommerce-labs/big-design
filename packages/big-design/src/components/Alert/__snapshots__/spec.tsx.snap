// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Alert 1`] = `
.c5 {
  vertical-align: middle;
  color: #3C64F4;
  height: 1.25rem;
  width: 1.25rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  border-radius: 0.25rem;
}

.c3 {
  padding-right: 0.75rem;
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c2 {
  border-left-width: 0.25rem;
  border-left-style: solid;
  border-left-color: #3C64F4;
}

@media (min-width:0px) {
  .c0 {
    padding: 0.5rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 0.75rem;
  }
}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1 c2"
>
  <div
    class="c3 c4"
  >
    <svg
      class="c5"
      color="primary"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
      />
    </svg>
  </div>
  <div
    class="c6 c7"
  >
    Dolore proident eiusmod sint est enim laboris anim minim quis ut adipisicing consectetur officia ex. Ipsum eiusmod fugiat amet pariatur culpa tempor aliquip tempor nisi. Irure esse deserunt nostrud ipsum id adipisicing enim velit labore. Nulla exercitation laborum laboris Lorem irure sit esse nulla mollit aliquip consectetur velit
  </div>
</div>
`;

exports[`render custom icon left alert 1`] = `
.c5 {
  vertical-align: middle;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  border-radius: 0.25rem;
}

.c3 {
  padding-right: 0.75rem;
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c2 {
  border-left-width: 0.25rem;
  border-left-style: solid;
  border-left-color: #3C64F4;
}

@media (min-width:0px) {
  .c0 {
    padding: 0.5rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 0.75rem;
  }
}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1 c2"
>
  <div
    class="c3 c4"
  >
    <svg
      class="c5"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v10zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
      />
    </svg>
  </div>
  <div
    class="c6 c7"
  >
    Dolore proident eiusmod sint est enim laboris
  </div>
</div>
`;

exports[`render danger alert 1`] = `
.c5 {
  vertical-align: middle;
  color: #DB3643;
  height: 1.25rem;
  width: 1.25rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  border-radius: 0.25rem;
}

.c3 {
  padding-right: 0.75rem;
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c2 {
  border-left-width: 0.25rem;
  border-left-style: solid;
  border-left-color: #DB3643;
}

@media (min-width:0px) {
  .c0 {
    padding: 0.5rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 0.75rem;
  }
}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1 c2"
>
  <div
    class="c3 c4"
  >
    <svg
      class="c5"
      color="danger"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 11c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v4c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
      />
    </svg>
  </div>
  <div
    class="c6 c7"
  >
    Dolore proident eiusmod sint est enim laboris
  </div>
</div>
`;

exports[`render success alert 1`] = `
.c5 {
  vertical-align: middle;
  color: #2AAB3F;
  height: 1.25rem;
  width: 1.25rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  border-radius: 0.25rem;
}

.c3 {
  padding-right: 0.75rem;
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c2 {
  border-left-width: 0.25rem;
  border-left-style: solid;
  border-left-color: #2AAB3F;
}

@media (min-width:0px) {
  .c0 {
    padding: 0.5rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 0.75rem;
  }
}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1 c2"
>
  <div
    class="c3 c4"
  >
    <svg
      class="c5"
      color="success"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.29 16.29L5.7 12.7a.996.996 0 111.41-1.41L10 14.17l6.88-6.88a.996.996 0 111.41 1.41l-7.59 7.59a.996.996 0 01-1.41 0z"
      />
    </svg>
  </div>
  <div
    class="c6 c7"
  >
    Dolore proident eiusmod sint est enim laboris
  </div>
</div>
`;

exports[`render warning alert 1`] = `
.c5 {
  vertical-align: middle;
  color: #FFBF00;
  height: 1.25rem;
  width: 1.25rem;
}

.c0 {
  box-sizing: border-box;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-shadow: 0px 1px 6px rgba(49,52,64,0.2);
  border-radius: 0.25rem;
}

.c3 {
  padding-right: 0.75rem;
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 {
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c2 {
  border-left-width: 0.25rem;
  border-left-style: solid;
  border-left-color: #FFBF00;
}

@media (min-width:0px) {
  .c0 {
    padding: 0.5rem;
  }
}

@media (min-width:720px) {
  .c0 {
    padding: 0.75rem;
  }
}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1 c2"
>
  <div
    class="c3 c4"
  >
    <svg
      class="c5"
      color="warning"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M4.47 21h15.06c1.54 0 2.5-1.67 1.73-3L13.73 4.99c-.77-1.33-2.69-1.33-3.46 0L2.74 18c-.77 1.33.19 3 1.73 3zM12 14c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
      />
    </svg>
  </div>
  <div
    class="c6 c7"
  >
    Dolore proident eiusmod sint est enim laboris
  </div>
</div>
`;

exports[`renders a header and action 1`] = `
.c3 {
  vertical-align: middle;
  height: 1rem;
  width: 1rem;
}

.c0 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: background-color,border-color,box-shadow,color;
  transition-property: background-color,border-color,box-shadow,color;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #D9DCE9;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  font-size: 1rem;
  font-weight: 400;
  height: 2.25rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 2rem;
  outline: none;
  padding: 0 1rem;
  pointer-events: auto;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
  padding: 0 0.5rem;
  background-color: transparent;
  border-color: transparent;
  color: #3C64F4;
}

.c0:focus {
  outline: none;
}

.c0[disabled] {
  border-color: #D9DCE9;
  pointer-events: none;
}

.c0 + .bd-button {
  margin-top: 0.5rem;
}

.c0:active {
  background-color: #DBE3FE;
}

.c0:focus {
  box-shadow: 0 0 0 0.25rem #DBE3FE;
}

.c0:hover:not(:active) {
  background-color: #F0F3FF;
}

.c0[disabled] {
  border-color: transparent;
  color: #D9DCE9;
}

.c2 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-grid;
  grid-auto-flow: column;
  grid-gap: 0.5rem;
  visibility: visible;
}

.c1 {
  color: #313440;
  height: 1.75rem;
  width: auto;
}

@media (min-width:720px) {
  .c0 + .bd-button {
    margin-top: 0;
    margin-left: 0.5rem;
  }
}

@media (min-width:720px) {
  .c0 {
    width: auto;
  }
}

<button
  class="c0 c1"
  role="button"
  tabindex="0"
>
  <span
    class="c2"
  >
    <svg
      class="c3"
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M18.3 5.71a.996.996 0 00-1.41 0L12 10.59 7.11 5.7A.996.996 0 105.7 7.11L10.59 12 5.7 16.89a.996.996 0 101.41 1.41L12 13.41l4.89 4.89a.996.996 0 101.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
      />
    </svg>
  </span>
</button>
`;
